pennylane/transforms/core/transform_program.py

```diff
@@ -167,6 +167,61 @@ def insert_front(self, transform_container: TransformContainer):
             )
         self._transform_program.insert(0, transform_container)
 
+    def add_transform(self, transform: TransformDispatcher, *targs, **tkwargs):
+        """Add a transform (dispatcher) to the end of the program.
+
+        Args:
+            transform(TransformDispatcher): The transform program where the transform is added.
+            *targs: Any additional arguments that are passed to the transform.
+
+        Keyword Args:
+            **tkwargs: Any additional keyword arguments that are passed to the transform.
+
+        """
+        if not isinstance(transform, TransformDispatcher):
+            raise TransformError("Only transform dispatcher can be added to the transform program.")
+
+        if transform.expand_transform:
+            self.push_back(TransformContainer(transform.expand_transform))
+        self.push_back(
+            TransformContainer(
+                transform.transform,
+                targs,
+                tkwargs,
+                transform.classical_cotransform,
+                transform.is_informative,
+            )
+        )
+
+    def insert_front_transform(self, transform: TransformDispatcher, *targs, **tkwargs):
+        """Add a transform (dispatcher) to the beginning of the program.
+
+        Args:
+            transform(TransformDispatcher): The transform program where the transform is added.
+            *targs: Any additional arguments that are passed to the transform.
+
+        Keyword Args:
+            **tkwargs: Any additional keyword arguments that are passed to the transform.
+
+        """
+        if transform.is_informative and not self.is_empty():
+            raise TransformError(
+                "Informative transforms can only be added at the end of the program."
```

Comment:
why is this the case?

Reply:
If they try add let's say qml.matrix at the beginning of the program.